### ISSUE TITLE
No `location` on Node 19

### ISSUE THREAD

**Issue by regseb** [prantlf/jsonlint#21]
In Node 19, the throwed error no longer contains the `location` (and the `pointer`) property.

## To reproduce

```JavaScript
import { parse } from "@prantlf/jsonlint";

console.log("Node " + process.version);
try {
    parse('{ "foo": baz }');
} catch (err) {
    console.log(err);
}
```

### In Node 18

```
Node v18.16.0
SyntaxError: Parse error on line 1, column 10:
{ "foo": baz }
---------^
Unexpected token b
    at JSON.parse (<anonymous>)
    at parseNative (/home/regseb/testcase/node_modules/@prantlf/jsonlint/lib/jsonlint.js:941:17)
    at parse (/home/regseb/testcase/node_modules/@prantlf/jsonlint/lib/jsonlint.js:970:7)
    at file:///home/regseb/testcase/testcase.js:5:5
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25) {
  reason: 'Unexpected token b',
  excerpt: '{ "foo": baz }',
  pointer: '---------^',
  location: { start: { column: 10, line: 1, offset: 9 } }
}
```

### In Node 19

```
Node v19.8.1
SyntaxError: Parse error in JSON input:
{ "foo": baz }
Unexpected token 'b', "{ "foo": baz }" is not valid JSON
    at JSON.parse (<anonymous>)
    at parseNative (/workspaces/testcase/node_modules/@prantlf/jsonlint/lib/jsonlint.js:941:17)
    at parse (/workspaces/testcase/node_modules/@prantlf/jsonlint/lib/jsonlint.js:970:7)
    at file:///workspaces/testcase/testcase.js:5:5
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25) {
  reason: `Unexpected token 'b', "{ "foo": baz }" is not valid JSON`,
  excerpt: '{ "foo": baz }'
}
```

**Comment by prantlf** [comment]
Thanks for the bug report! Yes, it appears that reading the error location from the native `JSON.parse`'s error message is not always reliable. The reliability appears to depend on the type of the error. And the results with the V8 from Node.js 19 are different than from Node.js 18, which remained the same since Node.js 16.

The failure from parsing `{ \"foo\": \"baz }` didn't contain the location, but from `{ \"foo\": baz }` did:

    ❯ node --version
    v18.16.0
    
    ❯ node -e 'try { JSON.parse("{ \"foo\": \"baz }") } catch (e) { console.log(e.message) }'
    Unexpected end of JSON input
    
    ❯ node -e 'try { JSON.parse("{ \"foo\": baz }") } catch (e) { console.log(e.message) }'
    Unexpected token b in JSON at position 9

The situation turned over with Node.js 19 and 20, but there's still one unreliable case:

    ❯ node --version
    v19.9.0
    
    ❯ node -e 'try { JSON.parse("{ \"foo\": \"baz }") } catch (e) { console.log(e.message) }'
    Unterminated string in JSON at position 15
    
    ❯ node -e 'try { JSON.parse("{ \"foo\": baz }") } catch (e) { console.log(e.message) }'
    Unexpected token 'b', "{ "foo": baz }" is not valid JSON
    
I use the native `JSON.parse` if a plain JSON string is parsed, if no features like CJSON or JSON5 are requested, because it's a lot faster. But the unreliability of the error location is unpleasant. I think that if the native parsing fails and there's no location available in the error message, I could repeat the parsing with the custom parser, which produces the error location reliably. JSON files are usually valid. Slowing down the operation in case of an error should be acceptable.

**Comment by prantlf** [comment]
The error output once more with the current `jsonlint`. It anticipates the missing locatoin and uses two different message prefixes:

    'Parse error on line ' + line + ', column ' + ':'
    'Parse error in JSON input:'

So, the messages have been already inconsistent since I introduced the optional native parsing. It appears that the new error handling in V8 included in Node.js 19 makes the other case happen more often:

    ❯ node --version
    v19.9.0
    
    ❯ node -e 'try { require("@prantlf/jsonlint").parse("{ \"foo\": \"baz }") } catch (e) { console.log(e.message) }'
    Parse error on line 1, column 16:
    { "foo": "baz }
    ---------------^
    Unterminated string

    ❯ node -e 'try { require("@prantlf/jsonlint").parse("{ \"foo\": baz }") } catch (e) { console.log(e.message) }'
    Parse error in JSON input:
    { "foo": baz }
    Unexpected token 'b', "{ "foo": baz }" is not valid JSON

The same with the custom parser enforced:
    
    ❯ node -e 'try { require("@prantlf/jsonlint").parse("{ \"foo\": \"baz }", { ignoreComments: true }) } catch (e) { console.log(e.message) }'
    Parse error on line 1, column 16:
    { "foo": "baz }
    ---------------^
    Unexpected end of input

    ❯ node -e 'try { require("@prantlf/jsonlint").parse("{ \"foo\": baz }", { ignoreComments: true }) } catch (e) { console.log(e.message) }'
    Parse error on line 1, column 10:
    { "foo": baz }
    ---------^
    No value found for key "foo"

**Comment by regseb** [comment]
The interest of `@prantlf/jsonlint` is to have detailed errors. If a user accepts to have simplistic errors, he can directly use `JSON.parse()`.

It's a good idea to pre-validate the JSON with `JSON.parse()`:

```JavaScript
export const parse = (input) => {
    try {
        return JSON.parse(input);
    } catch {
        return parseCustom(input);
    }
};
```

**Comment by prantlf** [comment]
Fixed by in [14.0.3](https://github.com/prantlf/jsonlint/releases/tag/v14.0.3) by [9757213](https://github.com/prantlf/jsonlint/commit/9757213eda5de9684099024d0c4f59e4d4f59c97).
